src/FlatSharp.Compiler/FlatSharpCompiler.cs

```diff
@@ -263,25 +264,53 @@ internal static byte[] GetBfbs(CompilerOptions options)
             Directory.CreateDirectory(outputDir);
             FileInfo info = new FileInfo(options.InputFile);
 
-            System.Diagnostics.Process p = new System.Diagnostics.Process
+            using var p = new Process
             {
-                StartInfo = new System.Diagnostics.ProcessStartInfo
+                StartInfo = new ProcessStartInfo
                 {
                     CreateNoWindow = true,
                     RedirectStandardError = true,
                     RedirectStandardOutput = true,
                     UseShellExecute = false,
                     FileName = flatcPath,
-                    Arguments = $"-b --schema --bfbs-comments --bfbs-builtins --bfbs-filenames \"{info.DirectoryName}\" --no-warnings -o \"{outputDir}\" \"{info.FullName}\"",
                 }
             };
 
+            foreach (var arg in new[]
```

Comment:
I addressed this with f263635ad2937da1ffcec66c790d1bffc935621b and using the dammit operator (`!`) on the `info.DirectoryName` that seemed to be the root of the issue. Unfortunately, [`FileInfo.DirectoryName`](https://docs.microsoft.com/en-us/dotnet/api/system.io.fileinfo.directoryname?view=net-6.0) is typed to return `string?` and this was causing the type of the whole array to be inferred as `string?[]`, and causing a warning to be issued later when adding each item to the list (where the list is typed to receive non-nullable strings). Rather than assert, I've used the dammit operator (`!`) to fix the issue along with a comment because, technically speaking, a file always lives in a directory and therefore I can't imagine `FileInfo.DirectoryName` ever returning a `null`. The fact that it's typed as `string?` is odd and I think comes from [its internal use of `Path.GetDirectoryName`](https://github.com/dotnet/runtime/blob/12a8819eee9865eb38bca6c05fdece1053102854/src/libraries/System.Private.CoreLib/src/System/IO/FileInfo.cs#L45), which can return `null` if the path is null/empty/root.

I find making sure that the array type is accurate is the right call here. If you feel strongly about asserting rather than using `!` then I can apply the following change:

```diff
diff --git a/src/FlatSharp.Compiler/FlatSharpCompiler.cs b/src/FlatSharp.Compiler/FlatSharpCompiler.cs
index 33e99ab..a5341cc 100644
--- a/src/FlatSharp.Compiler/FlatSharpCompiler.cs
+++ b/src/FlatSharp.Compiler/FlatSharpCompiler.cs
@@ -276,6 +276,9 @@ namespace FlatSharp.Compiler
                 }
             };
 
+            var inputDir = info.DirectoryName;
+            FlatSharpInternal.Assert(inputDir is not null, "The file directory was null.");
+
             foreach (var arg in new[]
             {
                 "-b",
@@ -283,7 +286,7 @@ namespace FlatSharp.Compiler
                 "--bfbs-comments",
                 "--bfbs-builtins",
                 "--bfbs-filenames",
-                info.DirectoryName!, // Files always have a directory name, dammit!
+                inputDir,
                 "--no-warnings",
                 "-o",
                 outputDir,
```